rpc/src/main/scala/proto/converters.scala

```diff
@@ -0,0 +1,194 @@
+        }
+      }
+  }
+
+  trait ScalaMetaTrait2ProtoService {
+    def convert(t: Trait): ProtoService
+  }
+
+  object ScalaMetaTrait2ProtoService {
+    implicit def defaultTrait2ServiceConverter =
+      new ScalaMetaTrait2ProtoService {
+        override def convert(t: Trait): ProtoService = {
+          ProtoService(
+            name = t.name.value,
+            rpcs = t.collect {
+              case q"@rpc @stream[ResponseStreaming] def $name[..$tparams]($request, observer: StreamObserver[$response]): FS[Unit]" =>
+                ProtoServiceField(
+                  name = name.value,
+                  request = extractParamType(request),
+                  response = response.toString(),
+                  streamingType = Some(ResponseStreaming))
+              case q"@rpc @stream[RequestStreaming] def $name[..$tparams]($param): FS[StreamObserver[$response]]" =>
+                ProtoServiceField(
+                  name = name.value,
+                  request = extractParamStreamingType(param),
+                  response = response.toString(),
+                  streamingType = Some(RequestStreaming)
+                )
+              case q"@rpc @stream[BidirectionalStreaming] def $name[..$tparams]($param): FS[StreamObserver[$response]]" =>
+                ProtoServiceField(
+                  name = name.value,
+                  request = extractParamStreamingType(param),
+                  response = response.toString(),
+                  streamingType = Some(BidirectionalStreaming)
+                )
+              case q"@rpc def $name[..$tparams]($request): FS[$response]" =>
+                ProtoServiceField(
+                  name = name.value,
+                  request = extractParamType(request),
+                  response = response.toString(),
+                  streamingType = None)
+            }
+          )
+        }
+
+        private[this] def extractParamType(param: Term.Param): String =
+          param.decltpe match {
+            case Some(retType) => retType.toString()
+            case None =>
+              throw new IllegalArgumentException(s"unexpected $param without return type")
+          }
+
+        private[this] def extractParamStreamingType(param: Term.Param): String =
+          param.decltpe match {
+            case Some(retType) =>
+              retType match {
+                case t"StreamObserver[$request]" => request.toString()
+                case _ =>
+                  throw new IllegalArgumentException(s"$param not enclosed in StreamObserver[_]")
+              }
+            case None =>
+              throw new IllegalArgumentException(s"unexpected $param without return type")
+          }
+
+      }
+  }
+
+  trait ScalaMetaObject2ProtoOptions {
+    def convert(o: Object): List[ProtoOption]
+  }
+
+  object ScalaMetaObject2ProtoOptions {
+    implicit def defaultObject2Options: ScalaMetaObject2ProtoOptions =
+      new ScalaMetaObject2ProtoOptions {
```

Comment:
It could be simplified to

```scala
object ScalaMetaObject2ProtoOptions {
    implicit def defaultObject2Options: ScalaMetaObject2ProtoOptions =
      (o: Object) => {
        o.mods.collect {
          case Mod.Annot(
              Term.Apply(
                Ctor.Ref.Name("option"),
                Seq(
                  Term.Arg.Named(Term.Name("name"), Lit.String(name)),
                  Term.Arg.Named(Term.Name("value"), Lit.String(value)),
                  Term.Arg.Named(Term.Name("quote"), Lit.Boolean(quote))))) =>
            ProtoOption(name, value, quote)
        }.toList
      }
  }
```